packages/core/src/storages/sitemap_request_list.ts

```diff
@@ -0,0 +1,103 @@
+import { parseSitemap } from '@crawlee/utils';
+
+import type { IRequestList } from './request_list';
+import { Request } from '../request';
+
+export class SitemapRequestList implements IRequestList {
+    inProgress = new Set<string>();
+
+    reclaimed = new Set<string>();
+
+    requests: Request[] = [];
+
+    uniqueKeyToIndex = new Map<string, number>();
+
+    private nextIndex = 0;
+
+    private isSitemapFullyLoaded = false;
+
+    private constructor(...args: Parameters<typeof parseSitemap>) {
+        (async () => {
+            for await (const item of parseSitemap(args[0], args[1])) {
+                this.addRequest(item.url);
+            }
+        })()
+            .then(() => {
+                this.isSitemapFullyLoaded = true;
+            })
+            .catch(() => {
+                this.isSitemapFullyLoaded = true;
+            });
+    }
+
+    static async open({ sitemapUrls }: { sitemapUrls: string | string[] }): Promise<SitemapRequestList> {
+        return new SitemapRequestList(
+            (Array.isArray(sitemapUrls) ? sitemapUrls : [sitemapUrls]).map((url) => ({ type: 'url', url })),
+        );
+    }
+
+    length(): number {
+        return this.requests.length;
+    }
+
+    async isFinished(): Promise<boolean> {
+        return this.inProgress.size === 0 && this.nextIndex >= this.requests.length && this.isSitemapFullyLoaded;
+    }
+
+    async isEmpty(): Promise<boolean> {
+        return this.reclaimed.size === 0 && this.nextIndex >= this.requests.length;
+    }
+
+    handledCount(): number {
+        return this.nextIndex - this.inProgress.size;
+    }
+
+    async persistState(): Promise<void> {
+        throw new Error('SitemapRequestList persistence is not yet implemented.');
```

Comment:
i think this is pretty important to support, i wouldn't merge it without it personally